src/test/java/org/wildfly/bot/PRSkipPullRequestTest.java

```diff
@@ -81,4 +95,49 @@ void testSkippingFormatCheckOnDraft() throws Throwable {
                     verifyNoMoreInteractions(mocks.pullRequest(pullRequestJson.id()));
                 });
     }
-}
+
+    @Test
+    void testSkippingRulesOnIncorrectRebase() throws Throwable {
+        final String duplicateSHA = "sha1";
+        final String baseBranch = "main";
+        pullRequestJson = TestModel.setPullRequestJsonBuilder(pullRequestJsonBuilder -> pullRequestJsonBuilder);
+        mockedContext = MockedGHPullRequest.builder(pullRequestJson.id());
+        TestModel.given(
+                mocks -> {
+                    WildflyGitHubBotTesting.mockRepo(mocks, wildflyConfigFile, pullRequestJson, mockedContext);
+                    GHCommit commit1 = mock(GHCommit.class);
+                    GHCommit commit2 = mock(GHCommit.class);
+
+                    when(commit1.getSHA1()).thenReturn(duplicateSHA);
+                    when(commit2.getSHA1()).thenReturn("sha2");
+
+                    // Create a fake PagedIterable for repository commits.
+                    Set<GHCommit> fakeRepoCommitSet = Set.of(commit1, commit2);
+                    PagedIterable<GHCommit> fakeRepoCommits = mock(PagedIterable.class);
+                    GHCommitQueryBuilder mockedQueryBuilder = mock(GHCommitQueryBuilder.class);
+
+                    when(mocks.repository(TestConstants.TEST_REPO).queryCommits()).thenReturn(mockedQueryBuilder);
+                    when(mockedQueryBuilder.from(eq(baseBranch))).thenReturn(mockedQueryBuilder);
+                    when(mockedQueryBuilder.pageSize(anyInt())).thenReturn(mockedQueryBuilder);
+                    when(mockedQueryBuilder.list()).thenReturn(fakeRepoCommits);
+                    when(fakeRepoCommits.toSet()).thenReturn(fakeRepoCommitSet);
+
+                    GHPullRequestCommitDetail fakePRCommit = mock(GHPullRequestCommitDetail.class);
+                    PagedIterator<GHPullRequestCommitDetail> mockIterator = mock(PagedIterator.class);
+                    PagedIterable<GHPullRequestCommitDetail> fakePRCommits = mock(PagedIterable.class);
+
+                    when(fakePRCommit.getSha()).thenReturn(duplicateSHA);
+                    when(mockIterator.hasNext()).thenReturn(true, false);
+                    when(mockIterator.next()).thenReturn(fakePRCommit);
+                    when(mocks.pullRequest(pullRequestJson.id()).listCommits()).thenReturn(fakePRCommits);
+                    when(fakePRCommits._iterator(anyInt())).thenReturn(mockIterator);
+                })
+                .pullRequestEvent(pullRequestJson)
+                .then(mocks -> {
+                    verify(mocks.pullRequest(pullRequestJson.id())).getBase();
+                    verify(mocks.repository(TestConstants.TEST_REPO)).queryCommits();
+                    verify(mocks.pullRequest(pullRequestJson.id()), never()).addLabels(any(String[].class));
+                    verify(mocks.pullRequest(pullRequestJson.id()), never()).requestReviewers(anyList());
```

Comment:
```suggestion
                    verify(mocks.pullRequest(pullRequestJson.id()), never()).addLabels(any());
                    verify(mocks.pullRequest(pullRequestJson.id()), never()).requestReviewers(any());
```
`any()` also includes `null`, so I think it's better to have it for all cases when using `never()*` since it widens the set of rejected conditions.

> *there are exceptions, fore example:
```java
verify(code()).foo("expected-value"); // one expected value
verify(code(),never()).foo(any());  // all the other values are not allowed – this does not work because any() also accepts "expected-value"
```